### PR TITLE
Codacy integration

### PR DESCRIPTION
**Pull request by FedorSmirnov89**
Adjusted the travis.yml file to integrate codacy. Codacy is a CI - framework for a visualization of code quality and test coverage, very similar to Sonarqube. You can view the results at 

https://app.codacy.com/projects

As soon as we integrate it, we also will receive messages about the coverage and the code quality changes brought by pull requests (both here and in Slack). So it is like coveralls, but also provides a code quality analysis.

I am quite sure that my changes to the travis.yml get codacy to work and supply it with our coverage data. However, I have no idea how the whole Maven stuff works. It would be great if you would check whether there could be any problems.

I also added some test for the SelectMapGenotype while playing around.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
 dist: trusty
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install jq
+  - wget -O ~/codacy-coverage-reporter-assembly-latest.jar $(curl https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r .assets[0].browser_download_url)
+
 after_success:
-- "./gradlew coveralls"
+- gradle jacocoRootReport
```
Comment (felixreimann) [inline .travis.yml]: ./gradlew

File: build.gradle [hunk 2]
```diff
@@ -1,6 +1,4 @@
-plugins {
-	id 'com.github.kt3k.coveralls' version '2.6.3'
-}
+
```
Comment (felixreimann) [inline build.gradle]: delete empty line

File: .travis.yml (hunk 1 above)
Comment (FedorSmirnov89) [reply .travis.yml]: I have already set the github and the codacy tokens for the travis of opt4j. As I have write access to this repo, this should be enough, but we can also replace it by your gitHub token. One thing I cannot do is allowing codacy to have a hook for commits and pushes to the repo, as I am not the repository admin. Could you please do that? You just have to go to codacy, login with gitHub, go opt4j/settings/integrations and set the checkbox for Post-Commit Hook

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install jq
+  - wget -O ~/codacy-coverage-reporter-assembly-latest.jar $(curl https://api.github.com/repos/codacy/codacy-coverage-reporter/releases/latest | jq -r .assets[0].browser_download_url)
+
 after_success:
-    - "./gradlew coveralls"
+- "./gradlew coveralls"
+- gradlew jacocoRootReport
```
Comment (felixreimann) [inline .travis.yml]: please combine these two lines to "./gradlew jacocoRootReport coveralls"

**Pull request merged**